onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -1031,8 +1033,13 @@ struct ProviderLibrary {
     }
 
     provider_ = PGetProvider();
+    provider_->Initialize();
     return provider_;
   }
+  ORT_CATCH(...) {
+    Unload(); // If anything fails we unload the library and rethrow
```

Comment:
hm, there are other failure paths that don't go through here, for example:
[https://github.com/microsoft/onnxruntime/blob/cdc8137e2a827e0bb8cb8ef5b1cc05838599b7c9/onnxruntime/core/session/provider_bridge_ort.cc#L1032](https://github.com/microsoft/onnxruntime/blob/cdc8137e2a827e0bb8cb8ef5b1cc05838599b7c9/onnxruntime/core/session/provider_bridge_ort.cc#L1032)
should any clean up happen there?

Reply:
sounds good